doc/source/yt4differences.rst

```diff
@@ -23,60 +23,67 @@ Updating to yt 4.0 from Old Versions (and going back)
-in which colormaps with colliding names raise errors. The fix is to explicitly
-import the ``cmocean`` module and prefix ``cmocean`` colormaps (like
-``balance``) with ``cmo.``.  Note that this solution works with any
-yt-supported version of Matplotlib, but is not backward compatible with earlier
-versions of yt. Example:
-
-Old Script (will not work in the near future):
-
-.. code-block:: python
-
-    import yt
-
-    ds = yt.testing.fake_random_ds(1)
-    p = yt.SlicePlot(ds, "z", ("gas", "density"))
-    p.set_cmap(("gas", "density"), "balance")
-
-New Script:
-
-.. code-block:: python
-
-    import yt
-    import cmocean
-
-    ds = yt.testing.fake_random_ds(1)
-    p = yt.SlicePlot(ds, "z", ("gas", "density"))
-    p.set_cmap(("gas", "density"), "cmo.balance")
+After installing yt-4.0, you’ll want to change your old scripts in a few key
+ways. After accounting for the changes described in the list below, try
+running your script. If it still fails, the callback failures in python
+are fairly descriptive and it may be possible to deduce what remaining
+changes are necessary. If you continue to have trouble, please don’t hesitate
+to :ref:`request help <asking-for-help>`.
+
+The list below is arranged in order of most important changes to least
+important changes.
+
+* **Fields should be specified as tuples not as strings**
+  In the past, you could specify fields as strings like ``"density"``, but
+  with the growth of yt and its many derived fields, there can be sometimes
+  be overlapping field names (e.g., ``("gas", "density")`` and
+  ``("PartType0", "density")``, where yt doesn't know which to use.  To avoid
+  these issues moving forward, please explicitly specify the full tuple form
+  of all fields. Just search for all field accesses in your scripts, and
+  replace strings with tuples (often just prefix your string with ``("gas",``).
+  There is a compatibility rule in yt-4.0 to allow strings to continue to work
+  until yt-4.1, but you may get unexpected behavior.  See our :ref:`fields`,
+  and :ref:`available field list <available-fields>` documentation for more
+  information.
+* **Neutral ion fields changing format**
+  In previous versions, neutral ion fields were specified as
+  ``ELEMENT_number_density`` (e.g., ``H_number_density`` to represent H I
+  number density).  This led to a lot of confusion, because some people assumed
+  these fields were the total hydrogen density, not neutral hydrogen density.
+  In yt-4.0, we have resolved this issue by explicitly calling total hydrogen
+  number density ``H_nuclei_density`` and neutral hydrogen density
+  ``H_p0_number_density`` (where ``p0`` refers to plus 0 charge).  This syntax
+  follows the rule for other ions: H II = ``H_p1`` = ionized hydrogen.  Change
+  your scripts accordingly.  See
+  `YTEP-0003 <https://ytep.readthedocs.io/en/latest/YTEPs/YTEP-0003.html#molecular-and-atomic-species-names>`_
+  describing field names for more information.
```

Comment:
Wow, I'm surprised YTEP 0003 needs to be documented here. Wasn't it implemented in yt 3.6 already ?

Reply:
Was it?  I thought this was only on dev.  I'm not sure it's widely known that this has changed, but maybe I'm just checked out.

Reply:
Well the YTEP itself says that it was implemented and apparently the source file has been stable for over 5 years.

Reply:
Well, the change to deprecating `H_number_density` as a field is only very recent.  That's what this note is all about, so I'm just making sure people know that.  I've changed the reference to the fields docs rather than the ytep.

Reply:
excellent